controlplane/pkg/model/model.go

```diff
@@ -204,6 +207,7 @@ func (i *impl) DeleteClientConnection(connectionId string) {
 		i.Unlock()
 		return
 	}
+	clientConnection.ConnectionState = ClientConnection_Closed
```

Comment:
Is this an unrelated fix?

Reply:
This fixes concurrency issue, some tests failed on build.

Reply:
Fix for concurrency issue and moved marking of connection is closed into this method.